### PR TITLE
Add tasks for all students and groups

### PR DESCRIPTION
**Pull request by zannloo**
Update to CreateDeadline and CreateTodo so it can create deadline/todo for all students
or groups

CreateDeadline /CreateTod is suppose to:
- create deadline/todo for all students in group if user specifies "allStu"
and is in a group directory.
- create deadline/todo for all groups in root if user specifies "allGrp"
and is in a root directory.

### CODE REVIEW

File: src/test/java/seedu/address/logic/newcommands/CreateDeadlineCommandTest.java [hunk 1]
```diff
@@ -63,6 +65,71 @@ public void execute_deadlineForStudentAccepted_addSuccessful() throws InvalidPat
...
+
+        LocalDateTime duedate = LocalDateTime.parse("2023-12-03T23:59");
+        Deadline deadline = new Deadline("Assignment 3", duedate);
+
+        RelativePath path = new RelativePath("~/grp-001/");
+
+        CreateDeadlineCommand command = new CreateDeadlineCommand(path, deadline, "allStu");
+        State state = new State(currPath, root, new UserPrefs());
+
+        assertTrue(alice.checkDuplicates(deadline));
```
Comment (mingyuanc) [inline src/test/java/seedu/address/logic/newcommands/CreateDeadlineCommandTest.java:101]: Good way to check

File: src/test/java/seedu/address/logic/newcommands/CreateDeadlineCommandTest.java [hunk 2]
```diff
@@ -63,6 +65,71 @@ public void execute_deadlineForStudentAccepted_addSuccessful() throws InvalidPat
...
+
+    //    @Test
+    //    public void execute_deadlineForAllGroupsInRootAccepted_addSuccessful() throws InvalidPathException,
+    //            UnsupportedPathOperationException, CommandException {
+    //        AbsolutePath currPath = new AbsolutePath("~");
+    //        Map<Id, Student> studentMap = new HashMap<>();
+    //        List<Task> defaultTaskList = new ArrayList<>();
+    //        defaultTaskList.add(new Deadline("Assignment 1", LocalDateTime.parse("2023-12-03T23:59")));
+    //        TaskList taskList1 = new TaskList(defaultTaskList);
+    //        TaskList taskList2 = new TaskList(defaultTaskList);
```
Comment (mingyuanc) [inline src/test/java/seedu/address/logic/newcommands/CreateDeadlineCommandTest.java:113]: Why is this commented out?

**Review by mingyuanc — DISMISSED** [review]
Have some comments but overall looks good

**Review by NereusWB922 — CHANGES_REQUESTED** [review]
Need some changes on Parser

**Review by NereusWB922 — APPROVED** [review]

**Review by NgChunMan — APPROVED** [review]
LGTM.

**Pull request merged**